lang/cs/Org.Apache.REEF.Common/metrics/MetricsSystem/MetricsRecord.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using Org.Apache.REEF.Common.Metrics.Api;
+
+namespace Org.Apache.REEF.Common.Metrics.MetricsSystem
+{
+    /// <summary>
+    /// Default implementation of <see cref="IMetricsRecord"/>. This class 
+    /// maintains a collection of immutable metrics and tags.
+    /// </summary>
+    internal sealed class MetricsRecord : IMetricsRecord
+    {
+        /// <summary>
+        /// Constructor for the metric record.
+        /// </summary>
+        /// <param name="info">Meta-data info for the record.</param>
+        /// <param name="timeStamp">Unix time stamp for the record measured as total 
+        /// seconds relative to Jan 01 1970. (UTC)</param>
+        /// <param name="metrics">Collection of metrics in the record.</param>
+        /// <param name="tags">Collection of tags in the record.</param>
+        /// <param name="context">Context of the record.</param>
+        public MetricsRecord(IMetricsInfo info,
+            long timeStamp,
+            IEnumerable<IImmutableMetric> metrics,
+            IEnumerable<MetricsTag> tags,
+            string context)
+        {
+            if (info == null)
+            {
+                throw new MetricsException("Record info cannot be null");
```

Comment:
We would like all exception in metrics to inherit from same exception class to avoid the situation we faced wrt. group comm. in IMRU later on. But I hjave introduced the suggested exception as inner exception.